File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        
        int i = 0 ;
        int j = nums.size();
        unordered_map<int, int> umap;
       
        for (int i = 0 ;  ; i++){
            if(umap.count(target - nums[i])){
                return {i , umap[target-nums[i]]};
            }
            else
            umap[nums[i]] = i ;
        }
        
      
       return res ; 
    }
};

File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    /*for(let i = 0 ; i < nums.length-1 ; i++ ) {
        for(let j = i+1 ; j < nums.length ; j++){
            if(nums[i] + nums[j] == target) return [i , j];
        }
    }
    return -1; */
    
    let map = new Map();
    let res = [];
    for(let i = 0 ; i < nums.length ; i++){
        
        if(map.has(target-nums[i])){
            res.push(map.get(target-nums[i]));
            res.push(i);
            return res;
        }
        else{
            map.set(nums[i], i);
        }
    }
    
    
};